File: backend/tests/invoice_test.rb
require 'test/unit'
require_relative '../src/invoice'

class InvoiceTest < Test::Unit::TestCase

  def setup
    @invoice = Invoice.new({
      :billing_address => 'Sample Billing Address',
      :shipping_address => 'Sample Shipping Address',
      :order => 'Sample Order'
    })
  end

  def test_can_create_invoice
    assert_instance_of(Invoice, @invoice)
  end

  def test_can_read_invoice_data
    assert_equal('Sample Billing Address', @invoice.billing_address.first)
    assert_equal('Sample Shipping Address', @invoice.shipping_address.first)
    assert_equal('Sample Order', @invoice.order.first)
  end
end

File: backend/src/test/kotlin/challenge/entity/InvoiceTest.kt
package challenge.entity

import org.junit.jupiter.api.Test
import kotlin.test.junit5.JUnit5Asserter

internal class InvoiceTest : BaseTest() {

    @Test
    fun `Assert billing address from invoice equals address from order`() {
        JUnit5Asserter.assertSame(expected = super.order.address, actual = super.invoice.billingAddress, message = null)
    }

    @Test
    fun `Assert shipping sddress from invoice equals address from order`() {
        JUnit5Asserter.assertSame(expected = super.order.address, actual = super.invoice.shippingAddress, message = null)
    }

    @Test
    fun `Assert order from invoice equals order`() {
        JUnit5Asserter.assertSame(expected = super.order, actual = super.invoice.order, message = null)
    }

}
